dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -442,9 +442,10 @@ private static void copyEntryProperties(Entry detachedEntry, Entry entry) {
         detachedEntry.setAliases(entry.getAliases());
         detachedEntry.setLabels((SortedSet<Label>)entry.getLabels());
         detachedEntry.setCheckerWorkflow(entry.getCheckerWorkflow());
-        // This is some weird hack to always set the topic (which is either automatic or manual) into the ES topicAutomatic property for search table
+        // This is some weird hack to always set the topic (which is either automatic, AI, or manual) into the ES topicAutomatic property for search table
         // This is to avoid indexing both topicAutomatic and topicManual and having the frontend choose which one to display
         detachedEntry.setTopicAutomatic(entry.getTopic());
+        detachedEntry.setTopicSelection(entry.getTopicSelection());
```

Comment:
Will this cause the topic selection to be part of the search? In other words, if I search for `automatic` in the UI, will all workflows whose topic selection is `automatic` match?

If so, it's minor, but it also doesn't seem like we'd want that. Is that perhaps controllable via the UI, e.g., set the weight to 0 for that field?

Reply:
In terms of highlighting, only fields specified here are used for Elasticsearch highlighting: 
https://github.com/dockstore/dockstore-ui2/blob/8a1faa120c58863192dfb7c0c2d4e181e0d296a3/src/app/search/query-builder.service.ts#L134-L145

As for the search results, the search queries and weights are specified here:
https://github.com/dockstore/dockstore-ui2/blob/8a1faa120c58863192dfb7c0c2d4e181e0d296a3/src/app/search/query-builder.service.ts#L277-L286

so to my understanding, as long as `topicSelection` isn't in any of those parts of the UI, it shouldn't be weighted in search results. It would be good to double check though

Reply:
> Will this cause the topic selection to be part of the search? In other words, if I search for `automatic` in the UI, will all workflows whose topic selection is `automatic` match?

Nope, no problemo, the UI limits a search to certain fields:
https://github.com/dockstore/dockstore-ui2/blob/develop/src/app/search/query-builder.service.ts#L272

Reply:
SO FAST WE ARE

Reply:
Thanks, concerns addressed